protocol/message_processor.go

```diff
@@ -428,12 +428,23 @@ func messageSpecToWhisper(spec *encryption.ProtocolMessageSpec) (*types.NewMessa
 	newMessage = &types.NewMessage{
 		TTL:       whisperTTL,
 		Payload:   payload,
-		PowTarget: whisperPoW,
+		PowTarget: calculatePoW(payload),
 		PowTime:   whisperPoWTime,
 	}
 	return newMessage, nil
 }
 
+// calculatePoW returns the PoWTarget to be used.
+// We check the size and arbitrarely set it to a lower PoW if the packet is
+// greater than 50KB. We do this as the defaultPoW is to high for clients to send
+// large messages.
+func calculatePoW(payload []byte) float64 {
+	if len(payload) > largeSizeInBytes {
+		return whisperLargeSizePoW
+	}
+	return whisperDefaultPoW
+}
```

Comment:
I know this is necessary. But is this open to abuse? I could negate the PoW limitations applied to text by sending huge text messages. Potentially only theoretical.

Reply:
You could, but you would not have much benefit from it, as the message just won't be received by older clients and will be received by newer, which probably is not achieving much.

The bit that is open for abuse is the fact that we lowered the PoW for the whole fleet, so you could instead send  messages quicker than before, as they would need a lower PoW, and those would be received by newer clients, so spamming is easier with this changes.

Having said that, with the PoW we had before was already trivial sending many messages in quick succession (you can send 500 in a few seconds), so yes, it's making things worse, but things were not great before :) 
Spamming is something we haven't quite figure out, but before deploying this probably best to run some tests with the differences.

Reply:
Ok, thank you. That is a helpful insight.